src/Command/Ingester/LtiInstanceIngesterCommand.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/*
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Command\Ingester;
+
+use OAT\SimpleRoster\Csv\CsvReaderBuilder;
+use OAT\SimpleRoster\Entity\LtiInstance;
+use OAT\SimpleRoster\Repository\LtiInstanceRepository;
+use OAT\SimpleRoster\Storage\StorageRegistry;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Throwable;
+
+class LtiInstanceIngesterCommand extends AbstractCsvIngesterCommand
+{
+    public const NAME = 'roster:ingest:lti-instance';
+
+    /** @var LtiInstanceRepository */
+    private $ltiInstanceRepository;
+
+    public function __construct(
+        CsvReaderBuilder $csvReaderBuilder,
+        StorageRegistry $storageRegistry,
+        LtiInstanceRepository $ltiInstanceRepository
+    ) {
+        parent::__construct($csvReaderBuilder, $storageRegistry);
+
+        $this->ltiInstanceRepository = $ltiInstanceRepository;
+    }
+
+    protected function getIngesterCommandName(): string
+    {
+        return self::NAME;
+    }
+
+    protected function configure(): void
+    {
+        parent::configure();
+
+        $this->setDescription('LTI instance ingestion from various sources');
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        parent::initialize($input, $output);
+
+        $this->symfonyStyle->title('Simple Roster - LTI Instance Ingester');
+    }
+
+    /**
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
```

Comment:
I might be wrong, but we could put this method also in the Abstract class. EX:
```
protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $this->symfonyStyle->text($this->getStartComment());
         $this->symfonyStyle->newLine();

          try {
             $this->progressBar->start();

              $numberOfProcessedRows = 0;
             $persisted = false;
             foreach ($this->csvReader->getRecords() as $rawLineItem) {
                 $this->validateRow(
                     $rawLineItem,
                     ...$this->getCsvColumns()
                 );

                  $numberOfProcessedRows++;
                 $persisted = false;

                  $this->processRow($rawLineItem);

                  if ($numberOfProcessedRows % $this->batchSize === 0) {
                     if (!$this->isDryRun) {
                         $this->processBatch();
                     }

                      $persisted = true;

                      $this->progressBar->advance($this->batchSize);
                 }
             }

              if (!$this->isDryRun && !$persisted) {
                 $this-> processBatch();
             }

              $this->progressBar->finish();

              $this->symfonyStyle->newLine(2);

              $verificationCommentMessage = $this->getVerificationComment();

              if ($this->isDryRun) {
                 $this->symfonyStyle->warning(
                     $this->getSucessfulComment($numberOfProcessedRows)
                 );

                  $this->symfonyStyle->comment($verificationCommentMessage);

                  return 0;
             }

              $this->symfonyStyle->success(
                 $this->getSucessfulComment($numberOfProcessedRows)
             );

              $this->symfonyStyle->comment($verificationCommentMessage);
         } catch (Throwable $exception) {
             $this->symfonyStyle->error($exception->getMessage());

              return 1;
         }

          return 0;
     }
```

Reply:
The idea is good, unfortunately there are slight differences between the 4 commands if you check them, which would make the abstraction too complicated. Just one example: How do you define what to process in `processbatch()`?